src/jit/codegenarm64.cpp

```diff
@@ -2420,23 +2420,27 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             // Floating point divide never raises an exception
             genCodeForBinary(treeNode);     
         }
-        else  // integer divide operation
+        else  // an integer divide operation
         {
             GenTreePtr divisorOp = treeNode->gtGetOp2();
+            emitAttr   size      = EA_ATTR(genTypeSize(genActualType(treeNode->TypeGet())));
 
             // TODO-ARM64-CQ: Optimize a divide by power of 2 as we do for AMD64
             
             if (divisorOp->IsZero())
             {
-                emitJumpKind jmpEqual = genJumpKindForOper(GT_EQ, CK_SIGNED);
-                genJumpToThrowHlpBlk(jmpEqual, SCK_DIV_BY_ZERO);
-                // We don't need to generate the sdiv/udiv instruction
+                // We unconditionally throw a divide by zero exception
+                genJumpToThrowHlpBlk(EJ_jmp, SCK_DIV_BY_ZERO);
+
+                // We still need to produce a register result 
+                emit->emitIns_R_R(INS_mov, size, targetReg, REG_ZR);
```

Comment:
Div operation never produces a gc-ref. Besides both code emitted by emitins_R_R() and the code produced by genProduceReg() both are dead code and will never get executed.  From that perspective setting targetReg to zero isn't serving any purpose.

I understand genProduceReg() call is still required for updating the life of targetReg but i don't see why we need to emit an instruction to set targetReg to zero.  That was my point